### ISSUE TITLE
Many different bugs

### ISSUE THREAD

**Issue by Romz24** [TheIllusiveC4/Polymorph#93]
#### Versions (Be specific, do not write "latest"):
* Polymorph: v0.0.12
* Fabric: v0.11.7
#### Observed Behavior:
1. I constantly see recipes that shouldn't be there, in my case it's **tin nugget** that can't be crafted in the current situation
![1](https://user-images.githubusercontent.com/2458113/135145319-0815fd58-409b-438b-88d5-a2bad541c8c0.png)
2. When choosing a recipe, the panel is visually duplicated and it looks strange
![2](https://user-images.githubusercontent.com/2458113/135145722-5e3ef5ce-d882-44a4-b70d-3e07dfda7fe9.png)
3. When melting, you cannot choose the final result, it will always be the same.
It looks like this bug: #87
4. The crafting menu is displayed below the items from the REI
![3](https://user-images.githubusercontent.com/2458113/135146437-ed11058b-e084-44d2-be9e-d4aaa7d558c8.png)

#### Video
https://youtu.be/1fwziEGGwcg

**Comment by TheIllusiveC4** [comment]
Which version of Fabric API are you using? Have you updated to the latest?

**Comment by Romz24** [comment]
> Which version of Fabric API are you using? Have you updated to the latest?

I try to update all mods to the current version, and fabric-api are no exception.

```fabric-api-0.40.1+1.16.jar```

**Comment by Romz24** [comment]
> > Which version of Fabric API are you using? Have you updated to the latest?
> 
> I try to update all mods to the current version, and fabric api are no exception.
> 
> `fabric-api-0.40.1+1.16.jar`

**Comment by TheIllusiveC4** [comment]
Okay, hand me your mod list then. Make sure you delete any mods that are not involved in the issue so that I have a minimal workspace to look at.

**Comment by TheIllusiveC4** [comment]
Also, I have confirmed 2 and 4 and am working on solutions for those.

I cannot reproduce 1 and 3, so I will need more information such as the specific recipes/mods involved.

**Comment by Romz24** [comment]
Do you need to drop only links to curseforge or upload archive with all the mods?

**Comment by TheIllusiveC4** [comment]
I need the version numbers also, so you can just list out the mod names and their version numbers. Also, I want to re-iterate, not all of the mods, just the ones involved. So, Polymorph, Fabric API, and I see Tech Reborn in your example. Maybe one or two more mods if their recipes are also involved. If it's more than 5 mods, you need to trim it down.

**Comment by Romz24** [comment]
I use 72 mods, and I don't know which one is causing the bug. Finding a mod that causes a bug will be very difficult.

**Comment by Romz24** [comment]
The melting error seems to be related to the ```better-nether-5.0.7.jar``` mod

List mods:
```
better-nether-5.0.7.jar
Cardinal-Components-API-2.8.3.jar
cloth-config-4.11.26-fabric.jar
fabric-api-0.40.1+1.16.jar
fabric-language-kotlin-1.6.4+kotlin.1.5.30.jar
indrev-1.9.18-BETA.jar
polymorph-fabric-0.0.10-1.16.5.jar
RebornCore-1.16-4.7.3+build.136.jar
RoughlyEnoughItems-5.12.248.jar
TechReborn-1.16-3.8.4+build.236.jar
```

Download: https://www.solidfiles.com/v/dNB3Aq8B2qnkM

**Comment by Romz24** [comment]
@TheIllusiveC4 

UPD: I spent a huge amount of time to figure this out, but in the end I found mods that cause bugs.
I think that at least one bug occurs due to the ```better-nether-5.0.7.jar``` mod

Video: https://youtu.be/GBc0W-IKBLc
Download: https://www.solidfiles.com/v/PexRW5pmGdVXw

```
astromine-1.11.5+fabric-1.16.5.jar
better-nether-5.0.7.jar
Cardinal-Components-API-2.8.3.jar
cloth-config-4.11.26-fabric.jar
fabric-api-0.40.1+1.16.jar
fabric-language-kotlin-1.6.4+kotlin.1.5.30.jar
indrev-1.9.18-BETA.jar
polymorph-fabric-0.0.12-1.16.5.jar
RebornCore-1.16-4.7.3+build.136.jar
RoughlyEnoughItems-5.12.248.jar
TechReborn-1.16-3.8.4+build.236.jar
```

**Comment by TheIllusiveC4** [comment]
Thank you very much for your cooperation, I'll review everything later on and report back with an update as soon as I can.

**Comment by TheIllusiveC4** [comment]
Okay, I understand what's happening in 1 now. I have working solutions for 1, 2, and 4. Still reviewing 3.

**Comment by TheIllusiveC4** [comment]
All should be fixed in 0.0.13. Thank you for reporting these issues.

**Comment by Romz24** [comment]
Thank you, I'm glad you responded quickly.
